src/btree/bt_delete.c

```diff
@@ -626,20 +645,17 @@ __wt_delete_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref, WT_PAGE_DELE
         WT_STAT_CONN_DATA_INCR(session, cache_read_deleted_prepared);
 
     /*
-     * Give the page a modify structure and mark the page dirty if the tree isn't read-only. If the
-     * tree can be written, the page must be marked dirty: otherwise it can be discarded, and that
-     * will lose the truncate information if the parent page hasn't been reconciled since the
-     * truncation happened.
-     *
-     * If the tree cannot be written (checked in page-modify-set), we won't dirty the page. In this
-     * case the truncate information must have been read from the parent page's on-disk cell, so we
-     * can fetch it again if we discard the page and then reread it.
+     * Give the page a modify structure. We need it to remember that the page has been instantiated.
+     * We do not need to mark the page dirty here. (It used to be necessary because evicting a clean
+     * instantiated page would lose the delete information; but that is no longer the case.) Note
```

Comment:
Add a tombstone to row store doesn't mark the page dirty? I think we go through row_modify to do that and that should mark the page dirty too.

Reply:
For row-store it adds the updates to the update lists by hand rather than going through row_modify. That lets it avoid marking the page dirty. (The new VLCS code uses col_modify because the data structures involved are a lot more complicated.) Clarified in [5431108](https://github.com/wiredtiger/wiredtiger/pull/8188/commits/5431108959e98cdf9b3aa1b0fb0aa532b7ab0db8)